### PR TITLE
QCD Smearing Template

### PR DESCRIPTION
**Pull request by mkilpatr**
Here is a small smearing template that Dipika can use to write the python code for the Smearing.

### CODE REVIEW

File: python/modules/qcdSmearProducer.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        self.out = wrappedOutputTree
+	self.out.branch("origRes", "F", lenVar="nJet*nGenJet");
+	self.out.branch("jetFlav", "F");
+
+    def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        pass
+
+    def jetResFunction(self, jets, genjets):
+	res = jets.pt/genjets.pt
```
Comment (zhenbinwu) [inline python/modules/qcdSmearProducer.py]: So this is the jet response? It seems you are calculate each jet with each genjet? No need for matching?

File: python/modules/qcdSmearProducer.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	met       = Object(event,     self.metBranchName)
+	weight    = Object(event,     "genWeight")
+
+	for j in jets :
+        	self.out.fillBranch("jetFlav", j.partonFlavour)
+		for gj in genjets :
+			origRes_ = [ self.jetResFunction(j, gj) ]
+
+	self.out.fillBranch("origRes", origRes_)
+        return True
```
[the remark this answers is not in the document]
Comment (mkilpatr) [reply python/modules/qcdSmearProducer.py]: I don't get an error here. Maybe theres a hidden space

File: python/modules/qcdSmearProducer.py (hunk 1 above)
Comment (mkilpatr) [reply python/modules/qcdSmearProducer.py]: I agree there should be a matching of the reco and genjet before this value is calculated. I only included the function as an example for Dipika

Comment (zhenbinwu) [reply python/modules/qcdSmearProducer.py]: I see. Actually NanoAOD stored the index to genjet  as
"Jet_genJetIdx | Int_t(index to Genjet) | index of matched gen jet"

You can just call that for linking to genjet.

**Comment by mkilpatr** [comment]
Added better genjet matching

**Comment by mkilpatr** [comment]
Can you please commit this template for qcd smearing

**Comment by zhenbinwu** [comment]
The code seems not ready for production yet. Is it just as template for Dipika to continue work on?

**Comment by mkilpatr** [comment]
Yes it is just a template for dipika. That's why I also added my own
Stop0l_postproc_QCD.py to run it. It is separated from the
Stop0l_postproc.py code.

**Comment by zhenbinwu** [comment]
OK, merging it for Dipika to start with.

**Comment by mkilpatr** [comment]
She has grabbed the file from my repository. It would just be better if she could fork from the origin repository instead of mine. Thank you.

**Pull request merged**